---
title: "Data Import and Processing"
author: "Marie, Anna, Michel"
date: today
format: 
    html:
      toc: true
      self-contained: true
      highlight-style: github
      code-line-numbers: true
editor_options: 
    chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Methods

Spike-ins are synthetic RNA fragments of known sequence and concentration added to RNA-seq for calibration and control. The table `lexogen.tsv` contains the known concentrations [fmol/µL] and the the molar mass [g/mol] of the spike-ins used in the experiment. The table `PacBio_dev_team_revio_samples.xlsx` contains information about the quantity and the different mixes of the spike-ins used based on the experiment date and replicate. The mass concentration  [fg/µL] is linearly correlated to the length of the RNA fragment and is the product of these two quantities. It is plotted against the TPM measured from RNA sequencing for Illumina+Salmon and PacBio+Bambu data. Furthermore, the TPM from llumina+Salmon and PacBio+Bambu are plotted against each other.

## Load packages

```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(readxl)
library(viridis)
library(cowplot)
```


### Load data

```{r}
pacbio <- read_excel("PacBio_dev_team_revio_samples.xlsx")
lexogen <- read.table(file = "lexogen_master.tsv", sep = "\t", header=TRUE)
df_list_bambu <- readRDS("./bambu.rds")
df_list_salmon <- readRDS("./salmon.rds")
```

### Spike-ins
```{r}
spike_in_ids <- lexogen$transcript_id
pacbio[, c("Day of the protocol", "spike in used*")]

## Salmon
salmon_0.025_spike <- df_list_salmon$salmon_0.025_tpm_ctrl[spike_in_ids,]
salmon_0.05_spike <- df_list_salmon$salmon_0.05_tpm_ctrl[spike_in_ids,]
salmon_0.1_spike <- df_list_salmon$salmon_0.1_tpm_ctrl[spike_in_ids,]
salmon_0.2_spike <- df_list_salmon$salmon_0.2_tpm_ctrl[spike_in_ids,]
## Bambu
bambu_0.025_spike <- df_list_bambu$bambu_0.025_tpm_ctrl[spike_in_ids,]
bambu_0.05_spike <- df_list_bambu$bambu_0.05_tpm_ctrl[spike_in_ids,]
bambu_0.1_spike <- df_list_bambu$bambu_0.1_tpm_ctrl[spike_in_ids,]
bambu_0.2_spike <- df_list_bambu$bambu_0.2_tpm_ctrl[spike_in_ids,]
```

The spike ins used differed based on the days - on day 0 - e1 was used and on day 5 - e2. 

## Plotting

### NDR 0.025

#### Salmon/Bambu against real concentration

Here we assume that there is linear relationship between the mass concentration and the TPM. 

```{r}
salmon_0.025_spike_day_0 <- subset(salmon_0.025_spike, select = grepl("day0", names(salmon_0.025_spike)))
salmon_0.025_spike_day_5 <- subset(salmon_0.025_spike, select = grepl("day5", names(salmon_0.025_spike)))

# Look at the relative amount of spike-in added
pacbio[, c("Day of the protocol", "amount added")]

# For the day 0 replicate 3 - data is missing. I assume 2 was added.
salmon_0.025_spike_day_0 <- salmon_0.025_spike_day_0 / 2
salmon_0.025_spike_day_0$spike_in_conc <- lexogen$molarity_e1_fmoles_microliter * lexogen$mw_g_mol 

# For the day 5 replicates 2, 3 - 2. 
salmon_0.025_spike_day_5[, 2:3] <- salmon_0.025_spike_day_5[, 2:3] / 2
salmon_0.025_spike_day_5$spike_in_conc <- lexogen$molarity_e2_fmoles_microliter * lexogen$mw_g_mol



# Day 0
day0.rep1 <- ggplot(data = salmon_0.025_spike_day_0, mapping = aes(x = salmon_0.025_spike_day_0$spike_in_conc, y = salmon_0.025_spike_day_0$`0.025_salmon_day0-rep1`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep2 <- ggplot(data = salmon_0.025_spike_day_0, mapping = aes(x = salmon_0.025_spike_day_0$spike_in_conc, y = salmon_0.025_spike_day_0$`0.025_salmon_day0-rep2`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep3 <- ggplot(data = salmon_0.025_spike_day_0, mapping = aes(x = salmon_0.025_spike_day_0$spike_in_conc, y = salmon_0.025_spike_day_0$`0.025_salmon_day0-rep3`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
day5.rep1 <- ggplot(data = salmon_0.025_spike_day_5, mapping = aes(x = salmon_0.025_spike_day_5$spike_in_conc, y = salmon_0.025_spike_day_5$`0.025_salmon_day5-rep1`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep2 <- ggplot(data = salmon_0.025_spike_day_5, mapping = aes(x = salmon_0.025_spike_day_5$spike_in_conc, y = salmon_0.025_spike_day_5$`0.025_salmon_day5-rep2`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)


salmon_0.025_spike_day_5$highlight <- ifelse(rownames(salmon_0.025_spike_day_5) =="SIRV311" , "highlight", "normal")
day5.rep3 <- ggplot(data = salmon_0.025_spike_day_5, mapping = aes(x = salmon_0.025_spike_day_5$spike_in_conc, y = salmon_0.025_spike_day_5$`0.025_salmon_day5-rep3`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM against real concentration\nSalmon 0.025", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))

# Outlier day 0
salmon_0.025_spike_day_0[order(salmon_0.025_spike_day_0$`0.025_salmon_day0-rep1`, decreasing=T), ][1, ]
salmon_0.025_spike_day_0[order(salmon_0.025_spike_day_0$`0.025_salmon_day0-rep2`, decreasing=T), ][1:2, ]
salmon_0.025_spike_day_0[order(salmon_0.025_spike_day_0$`0.025_salmon_day0-rep3`, decreasing=T), ][1, ]

# Outlier day 5
salmon_0.025_spike_day_5[order(salmon_0.025_spike_day_5$`0.025_salmon_day5-rep1`, decreasing=T), ][1, ]
salmon_0.025_spike_day_5[order(salmon_0.025_spike_day_5$`0.025_salmon_day5-rep2`, decreasing=T), ][1, ]
salmon_0.025_spike_day_5[order(salmon_0.025_spike_day_5$`0.025_salmon_day5-rep3`, decreasing=T), ][1, ]
```

The correlations between the mass concentration and the TPM is linear. However, there is one clear outlier, one point with very high TPM and low concentration based on the day - `SIRV618` day 0 and `SIRV311` day 5. For day 0 rep 2 `SIRV311` is an additional outlier. 

```{r}
bambu_0.025_spike_day_0 <- subset(bambu_0.025_spike, select = grepl("day0", names(bambu_0.025_spike)))
bambu_0.025_spike_day_5 <- subset(bambu_0.025_spike, select = grepl("day5", names(bambu_0.025_spike)))

# Look at the relative amount of spike-in added
pacbio[, c("Day of the protocol", "amount added")]

# For the day 0 replicate 3 - data is missing. I assume 2 was added.
bambu_0.025_spike_day_0 <- bambu_0.025_spike_day_0 / 2
bambu_0.025_spike_day_0$spike_in_conc <- lexogen$molarity_e1_fmoles_microliter * lexogen$mw_g_mol

# For the day 5 replicates 2, 3 - 2. 
bambu_0.025_spike_day_5[, 2:3] <- bambu_0.025_spike_day_5[, 2:3] / 2
bambu_0.025_spike_day_5$spike_in_conc <- lexogen$molarity_e2_fmoles_microliter * lexogen$mw_g_mol

# Day 0 
day0.rep1 <- ggplot(data = bambu_0.025_spike_day_0, mapping = aes(x = bambu_0.025_spike_day_0$spike_in_conc, y = bambu_0.025_spike_day_0$`day0.rep1.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Concentration[fg/µL", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep2 <- ggplot(data = bambu_0.025_spike_day_0, mapping = aes(x = bambu_0.025_spike_day_0$spike_in_conc, y = bambu_0.025_spike_day_0$`day0.rep2.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep3 <- ggplot(data = bambu_0.025_spike_day_0, mapping = aes(x = bambu_0.025_spike_day_0$spike_in_conc, y = bambu_0.025_spike_day_0$`day0.rep3.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
day5.rep1 <- ggplot(data = bambu_0.025_spike_day_5, mapping = aes(x = bambu_0.025_spike_day_5$spike_in_conc, y = bambu_0.025_spike_day_5$`day5.rep1.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep2 <- ggplot(data = bambu_0.025_spike_day_5, mapping = aes(x = bambu_0.025_spike_day_5$spike_in_conc, y = bambu_0.025_spike_day_5$`day5.rep2.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep3 <- ggplot(data = bambu_0.025_spike_day_5, mapping = aes(x = bambu_0.025_spike_day_5$spike_in_conc, y = bambu_0.025_spike_day_5$`day5.rep3.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM against real concentration\nbambu 0.025", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```

There is some correlation between the mass concentration and TPM between concentration and TPM. 

#### Salmon against Bambu
```{r}
# Day 0
salmon_bambu <- data.frame(cbind(salmon_0.025_spike_day_0$`0.025_salmon_day0-rep1`, bambu_0.025_spike_day_0$day0.rep1.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep1 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.025_spike_day_0$`0.025_salmon_day0-rep2`, bambu_0.025_spike_day_0$day0.rep2.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep2 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.025_spike_day_0$`0.025_salmon_day0-rep3`, bambu_0.025_spike_day_0$day0.rep3.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep3 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
salmon_bambu <- data.frame(cbind(salmon_0.025_spike_day_0$`0.025_salmon_day0-rep1`, bambu_0.025_spike_day_0$day0.rep1.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep1 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.025_spike_day_0$`0.025_salmon_day0-rep2`, bambu_0.025_spike_day_0$day0.rep2.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep2 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.025_spike_day_0$`0.025_salmon_day0-rep3`, bambu_0.025_spike_day_0$day0.rep3.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep3 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM measure in different methods/nNDR 0.025 0.025", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```

The correlation between the TPM for Salmon and Bambu is very low.

### NDR 0.05
#### Salmon/Bambu against real concentration

```{r}
salmon_0.05_spike_day_0 <- subset(salmon_0.05_spike, select = grepl("day0", names(salmon_0.05_spike)))
salmon_0.05_spike_day_5 <- subset(salmon_0.05_spike, select = grepl("day5", names(salmon_0.05_spike)))

# Look at the relative amount of spike-in added
pacbio[, c("Day of the protocol", "amount added")]

# For the day 0 replicate 3 - data is missing. I assume 2 was added.
salmon_0.05_spike_day_0 <- salmon_0.05_spike_day_0 / 2
salmon_0.05_spike_day_0$spike_in_conc <- lexogen$molarity_e1_fmoles_microliter * lexogen$mw_g_mol

# For the day 5 replicates 2, 3 - 2. 
salmon_0.05_spike_day_5[, 2:3] <- salmon_0.05_spike_day_5[, 2:3] / 2
salmon_0.05_spike_day_5$spike_in_conc <- lexogen$molarity_e2_fmoles_microliter * lexogen$mw_g_mol

# Day 0
day0.rep1 <- ggplot(data = salmon_0.05_spike_day_0, mapping = aes(x = salmon_0.05_spike_day_0$spike_in_conc, y = salmon_0.05_spike_day_0$`0.05_salmon_day0-rep1`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep2 <- ggplot(data = salmon_0.05_spike_day_0, mapping = aes(x = salmon_0.05_spike_day_0$spike_in_conc, y = salmon_0.05_spike_day_0$`0.05_salmon_day0-rep2`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep3 <- ggplot(data = salmon_0.05_spike_day_0, mapping = aes(x = salmon_0.05_spike_day_0$spike_in_conc, y = salmon_0.05_spike_day_0$`0.05_salmon_day0-rep3`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
day5.rep1 <- ggplot(data = salmon_0.05_spike_day_5, mapping = aes(x = salmon_0.05_spike_day_5$spike_in_conc, y = salmon_0.05_spike_day_5$`0.05_salmon_day5-rep1`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep2 <- ggplot(data = salmon_0.05_spike_day_5, mapping = aes(x = salmon_0.05_spike_day_5$spike_in_conc, y = salmon_0.05_spike_day_5$`0.05_salmon_day5-rep2`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep3 <- ggplot(data = salmon_0.05_spike_day_5, mapping = aes(x = salmon_0.05_spike_day_5$spike_in_conc, y = salmon_0.05_spike_day_5$`0.05_salmon_day5-rep3`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM against real concentration\nSalmon 0.05", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```

```{r}
bambu_0.05_spike_day_0 <- subset(bambu_0.05_spike, select = grepl("day0", names(bambu_0.05_spike)))
bambu_0.05_spike_day_5 <- subset(bambu_0.05_spike, select = grepl("day5", names(bambu_0.05_spike)))

# Look at the relative amount of spike-in added
pacbio[, c("Day of the protocol", "amount added")]

# For the day 0 replicate 3 - data is missing. I assume 2 was added.
bambu_0.05_spike_day_0 <- bambu_0.05_spike_day_0 / 2
bambu_0.05_spike_day_0$spike_in_conc <- lexogen$molarity_e1_fmoles_microliter * lexogen$mw_g_mol

# For the day 5 replicates 2, 3 - 2. 
bambu_0.05_spike_day_5[, 2:3] <- bambu_0.05_spike_day_5[, 2:3] / 2
bambu_0.05_spike_day_5$spike_in_conc <- lexogen$molarity_e2_fmoles_microliter * lexogen$mw_g_mol

# Day 0 
day0.rep1 <- ggplot(data = bambu_0.05_spike_day_0, mapping = aes(x = bambu_0.05_spike_day_0$spike_in_conc, y = bambu_0.05_spike_day_0$`day0.rep1.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep2 <- ggplot(data = bambu_0.05_spike_day_0, mapping = aes(x = bambu_0.05_spike_day_0$spike_in_conc, y = bambu_0.05_spike_day_0$`day0.rep2.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep3 <- ggplot(data = bambu_0.05_spike_day_0, mapping = aes(x = bambu_0.05_spike_day_0$spike_in_conc, y = bambu_0.05_spike_day_0$`day0.rep3.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
day5.rep1 <- ggplot(data = bambu_0.05_spike_day_5, mapping = aes(x = bambu_0.05_spike_day_5$spike_in_conc, y = bambu_0.05_spike_day_5$`day5.rep1.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep2 <- ggplot(data = bambu_0.05_spike_day_5, mapping = aes(x = bambu_0.05_spike_day_5$spike_in_conc, y = bambu_0.05_spike_day_5$`day5.rep2.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep3 <- ggplot(data = bambu_0.05_spike_day_5, mapping = aes(x = bambu_0.05_spike_day_5$spike_in_conc, y = bambu_0.05_spike_day_5$`day5.rep3.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM against real concentration\nbambu 0.05", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```

#### Salmon against Bambu
```{r}
# Day 0
salmon_bambu <- data.frame(cbind(salmon_0.05_spike_day_0$`0.05_salmon_day0-rep1`, bambu_0.05_spike_day_0$day0.rep1.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep1 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.05_spike_day_0$`0.05_salmon_day0-rep2`, bambu_0.05_spike_day_0$day0.rep2.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep2 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.05_spike_day_0$`0.05_salmon_day0-rep3`, bambu_0.05_spike_day_0$day0.rep3.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep3 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
salmon_bambu <- data.frame(cbind(salmon_0.05_spike_day_0$`0.05_salmon_day0-rep1`, bambu_0.05_spike_day_0$day0.rep1.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep1 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.05_spike_day_0$`0.05_salmon_day0-rep2`, bambu_0.05_spike_day_0$day0.rep2.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep2 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.05_spike_day_0$`0.05_salmon_day0-rep3`, bambu_0.05_spike_day_0$day0.rep3.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep3 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM measure in different methods/nNDR 0.025 0.05", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```


### NDR 0.1
#### Salmon/Bambu against real concentration

```{r}
salmon_0.1_spike_day_0 <- subset(salmon_0.1_spike, select = grepl("day0", names(salmon_0.1_spike)))
salmon_0.1_spike_day_5 <- subset(salmon_0.1_spike, select = grepl("day5", names(salmon_0.1_spike)))

# Look at the relative amount of spike-in added
pacbio[, c("Day of the protocol", "amount added")]

# For the day 0 replicate 3 - data is missing. I assume 2 was added.
salmon_0.1_spike_day_0 <- salmon_0.1_spike_day_0 / 2
salmon_0.1_spike_day_0$spike_in_conc <- lexogen$molarity_e1_fmoles_microliter * lexogen$mw_g_mol

# For the day 5 replicates 2, 3 - 2. 
salmon_0.1_spike_day_5[, 2:3] <- salmon_0.1_spike_day_5[, 2:3] / 2
salmon_0.1_spike_day_5$spike_in_conc <- lexogen$molarity_e2_fmoles_microliter * lexogen$mw_g_mol

# Day 0
day0.rep1 <- ggplot(data = salmon_0.1_spike_day_0, mapping = aes(x = salmon_0.1_spike_day_0$spike_in_conc, y = salmon_0.1_spike_day_0$`0.1_salmon_day0-rep1`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep2 <- ggplot(data = salmon_0.1_spike_day_0, mapping = aes(x = salmon_0.1_spike_day_0$spike_in_conc, y = salmon_0.1_spike_day_0$`0.1_salmon_day0-rep2`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep3 <- ggplot(data = salmon_0.1_spike_day_0, mapping = aes(x = salmon_0.1_spike_day_0$spike_in_conc, y = salmon_0.1_spike_day_0$`0.1_salmon_day0-rep3`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
day5.rep1 <- ggplot(data = salmon_0.1_spike_day_5, mapping = aes(x = salmon_0.1_spike_day_5$spike_in_conc, y = salmon_0.1_spike_day_5$`0.1_salmon_day5-rep1`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep2 <- ggplot(data = salmon_0.1_spike_day_5, mapping = aes(x = salmon_0.1_spike_day_5$spike_in_conc, y = salmon_0.1_spike_day_5$`0.1_salmon_day5-rep2`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep3 <- ggplot(data = salmon_0.1_spike_day_5, mapping = aes(x = salmon_0.1_spike_day_5$spike_in_conc, y = salmon_0.1_spike_day_5$`0.1_salmon_day5-rep3`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM against real concentration\nSalmon 0.1", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```

```{r}
bambu_0.1_spike_day_0 <- subset(bambu_0.1_spike, select = grepl("day0", names(bambu_0.1_spike)))
bambu_0.1_spike_day_5 <- subset(bambu_0.1_spike, select = grepl("day5", names(bambu_0.1_spike)))

# Look at the relative amount of spike-in added
pacbio[, c("Day of the protocol", "amount added")]

# For the day 0 replicate 3 - data is missing. I assume 2 was added.
bambu_0.1_spike_day_0 <- bambu_0.1_spike_day_0 / 2
bambu_0.1_spike_day_0$spike_in_conc <- lexogen$molarity_e1_fmoles_microliter * lexogen$mw_g_mol

# For the day 5 replicates 2, 3 - 2. 
bambu_0.1_spike_day_5[, 2:3] <- bambu_0.1_spike_day_5[, 2:3] / 2
bambu_0.1_spike_day_5$spike_in_conc <- lexogen$molarity_e2_fmoles_microliter * lexogen$mw_g_mol

# Day 0 
day0.rep1 <- ggplot(data = bambu_0.1_spike_day_0, mapping = aes(x = bambu_0.1_spike_day_0$spike_in_conc, y = bambu_0.1_spike_day_0$`day0.rep1.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep2 <- ggplot(data = bambu_0.1_spike_day_0, mapping = aes(x = bambu_0.1_spike_day_0$spike_in_conc, y = bambu_0.1_spike_day_0$`day0.rep2.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep3 <- ggplot(data = bambu_0.1_spike_day_0, mapping = aes(x = bambu_0.1_spike_day_0$spike_in_conc, y = bambu_0.1_spike_day_0$`day0.rep3.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
day5.rep1 <- ggplot(data = bambu_0.1_spike_day_5, mapping = aes(x = bambu_0.1_spike_day_5$spike_in_conc, y = bambu_0.1_spike_day_5$`day5.rep1.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep2 <- ggplot(data = bambu_0.1_spike_day_5, mapping = aes(x = bambu_0.1_spike_day_5$spike_in_conc, y = bambu_0.1_spike_day_5$`day5.rep2.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep3 <- ggplot(data = bambu_0.1_spike_day_5, mapping = aes(x = bambu_0.1_spike_day_5$spike_in_conc, y = bambu_0.1_spike_day_5$`day5.rep3.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM against real concentration\nbambu 0.1", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```

#### Salmon against Bambu
```{r}
# Day 0
salmon_bambu <- data.frame(cbind(salmon_0.1_spike_day_0$`0.1_salmon_day0-rep1`, bambu_0.1_spike_day_0$day0.rep1.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep1 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.1_spike_day_0$`0.1_salmon_day0-rep2`, bambu_0.1_spike_day_0$day0.rep2.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep2 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.1_spike_day_0$`0.1_salmon_day0-rep3`, bambu_0.1_spike_day_0$day0.rep3.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep3 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
salmon_bambu <- data.frame(cbind(salmon_0.1_spike_day_0$`0.1_salmon_day0-rep1`, bambu_0.1_spike_day_0$day0.rep1.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep1 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.1_spike_day_0$`0.1_salmon_day0-rep2`, bambu_0.1_spike_day_0$day0.rep2.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep2 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.1_spike_day_0$`0.1_salmon_day0-rep3`, bambu_0.1_spike_day_0$day0.rep3.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep3 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM measure in different methods/nNDR 0.025 0.1", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```

### NDR 0.2
#### Salmon/Bambu against real concentration

```{r}
salmon_0.2_spike_day_0 <- subset(salmon_0.2_spike, select = grepl("day0", names(salmon_0.2_spike)))
salmon_0.2_spike_day_5 <- subset(salmon_0.2_spike, select = grepl("day5", names(salmon_0.2_spike)))

# Look at the relative amount of spike-in added
pacbio[, c("Day of the protocol", "amount added")]

# For the day 0 replicate 3 - data is missing. I assume 2 was added.
salmon_0.2_spike_day_0 <- salmon_0.2_spike_day_0 / 2
salmon_0.2_spike_day_0$spike_in_conc <- lexogen$molarity_e1_fmoles_microliter * lexogen$mw_g_mol

# For the day 5 replicates 2, 3 - 2. 
salmon_0.2_spike_day_5[, 2:3] <- salmon_0.2_spike_day_5[, 2:3] / 2
salmon_0.2_spike_day_5$spike_in_conc <- lexogen$molarity_e2_fmoles_microliter * lexogen$mw_g_mol

# Day 0
day0.rep1 <- ggplot(data = salmon_0.2_spike_day_0, mapping = aes(x = salmon_0.2_spike_day_0$spike_in_conc, y = salmon_0.2_spike_day_0$`0.2_salmon_day0-rep1`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep2 <- ggplot(data = salmon_0.2_spike_day_0, mapping = aes(x = salmon_0.2_spike_day_0$spike_in_conc, y = salmon_0.2_spike_day_0$`0.2_salmon_day0-rep2`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep3 <- ggplot(data = salmon_0.2_spike_day_0, mapping = aes(x = salmon_0.2_spike_day_0$spike_in_conc, y = salmon_0.2_spike_day_0$`0.2_salmon_day0-rep3`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
day5.rep1 <- ggplot(data = salmon_0.2_spike_day_5, mapping = aes(x = salmon_0.2_spike_day_5$spike_in_conc, y = salmon_0.2_spike_day_5$`0.2_salmon_day5-rep1`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep2 <- ggplot(data = salmon_0.2_spike_day_5, mapping = aes(x = salmon_0.2_spike_day_5$spike_in_conc, y = salmon_0.2_spike_day_5$`0.2_salmon_day5-rep2`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep3 <- ggplot(data = salmon_0.2_spike_day_5, mapping = aes(x = salmon_0.2_spike_day_5$spike_in_conc, y = salmon_0.2_spike_day_5$`0.2_salmon_day5-rep3`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM against real concentration\nSalmon 0.2", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```

```{r}
bambu_0.2_spike_day_0 <- subset(bambu_0.2_spike, select = grepl("day0", names(bambu_0.2_spike)))
bambu_0.2_spike_day_5 <- subset(bambu_0.2_spike, select = grepl("day5", names(bambu_0.2_spike)))

# Look at the relative amount of spike-in added
pacbio[, c("Day of the protocol", "amount added")]

# For the day 0 replicate 3 - data is missing. I assume 2 was added.
bambu_0.2_spike_day_0 <- bambu_0.2_spike_day_0 / 2
bambu_0.2_spike_day_0$spike_in_conc <- lexogen$molarity_e1_fmoles_microliter * lexogen$mw_g_mol

# For the day 5 replicates 2, 3 - 2. 
bambu_0.2_spike_day_5[, 2:3] <- bambu_0.2_spike_day_5[, 2:3] / 2
bambu_0.2_spike_day_5$spike_in_conc <- lexogen$molarity_e2_fmoles_microliter * lexogen$mw_g_mol

# Day 0 
day0.rep1 <- ggplot(data = bambu_0.2_spike_day_0, mapping = aes(x = bambu_0.2_spike_day_0$spike_in_conc, y = bambu_0.2_spike_day_0$`day0.rep1.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep2 <- ggplot(data = bambu_0.2_spike_day_0, mapping = aes(x = bambu_0.2_spike_day_0$spike_in_conc, y = bambu_0.2_spike_day_0$`day0.rep2.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day0.rep3 <- ggplot(data = bambu_0.2_spike_day_0, mapping = aes(x = bambu_0.2_spike_day_0$spike_in_conc, y = bambu_0.2_spike_day_0$`day0.rep3.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
day5.rep1 <- ggplot(data = bambu_0.2_spike_day_5, mapping = aes(x = bambu_0.2_spike_day_5$spike_in_conc, y = bambu_0.2_spike_day_5$`day5.rep1.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep2 <- ggplot(data = bambu_0.2_spike_day_5, mapping = aes(x = bambu_0.2_spike_day_5$spike_in_conc, y = bambu_0.2_spike_day_5$`day5.rep2.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

day5.rep3 <- ggplot(data = bambu_0.2_spike_day_5, mapping = aes(x = bambu_0.2_spike_day_5$spike_in_conc, y = bambu_0.2_spike_day_5$`day5.rep3.aligned.sorted`,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Concentration [fg/µL]", y = "TPM", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM against real concentration\nbambu 0.2", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```

#### Salmon against Bambu
```{r}
# Day 0
salmon_bambu <- data.frame(cbind(salmon_0.2_spike_day_0$`0.2_salmon_day0-rep1`, bambu_0.2_spike_day_0$day0.rep1.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep1 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 1", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.2_spike_day_0$`0.2_salmon_day0-rep2`, bambu_0.2_spike_day_0$day0.rep2.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep2 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 2", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.2_spike_day_0$`0.2_salmon_day0-rep3`, bambu_0.2_spike_day_0$day0.rep3.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day0.rep3 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 0 Rep 3", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

# Day 5
salmon_bambu <- data.frame(cbind(salmon_0.2_spike_day_0$`0.2_salmon_day0-rep1`, bambu_0.2_spike_day_0$day0.rep1.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep1 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 1", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.2_spike_day_0$`0.2_salmon_day0-rep2`, bambu_0.2_spike_day_0$day0.rep2.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep2 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 2", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

salmon_bambu <- data.frame(cbind(salmon_0.2_spike_day_0$`0.2_salmon_day0-rep3`, bambu_0.2_spike_day_0$day0.rep3.aligned.sorted))
colnames(salmon_bambu) <- c("salmon", "bambu")
day5.rep3 <- ggplot(data = salmon_bambu, mapping = aes(x = salmon, y = bambu,)) +
  geom_point() +
  labs(title = "Day 5 Rep 3", x = "Salmon [TPM]", y = "Bambu [TPM]", color = "DE") +
  scale_color_viridis_d() +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', formula= y~x)

title <- ggdraw() + draw_label("Spike-in TPM measure in different methods/nNDR 0.025 0.2", fontface='bold')
p <- plot_grid(day0.rep1, day0.rep2, day0.rep3, day5.rep1, day5.rep2, day5.rep3, ncol=3)
plot_grid(title, p, ncol=1, rel_heights=c(0.1, 1))
```
